R/RLearner_regr_gstat.R

```diff
@@ -0,0 +1,164 @@
+      makeUntypedLearnerParam(id = "merge"),
+      makeIntegerLearnerParam(id = "degree", default = 0),
+      makeLogicalLearnerParam(id = "vdist", default = FALSE),
+      # gstat::variogram params
+      makeNumericLearnerParam(id = "cutoff"), # FIXME default value
+      makeNumericLearnerParam(id = "width"),
+      makeNumericLearnerParam(id = "alpha", default = 0),
+      makeNumericLearnerParam(id = "beta.variogram", default = 0),
+      makeNumericLearnerParam(id = "tol.hor"),
+      makeNumericLearnerParam(id = "tol.ver"),
+      makeLogicalLearnerParam(id = "cressie", default = FALSE),
+      makeNumericLearnerParam(id = "dX", default = 0),
+      makeNumericLearnerParam(id = "boundaries", default = 0),
+      makeLogicalLearnerParam(id = "cloud", default = FALSE),
+      makeUntypedLearnerParam(id = "trend.beta", default = NULL),
+      makeIntegerLearnerParam(id = "debug.level", default = 1),
+      makeLogicalLearnerParam(id = "cross", default = TRUE),
+      makeLogicalLearnerParam(id = "map", default = FALSE),
+      makeLogicalLearnerParam(id = "projected", default = TRUE),
+      makeNumericLearnerParam(id = "lambda", default = 1.0),
+      makeLogicalLearnerParam(id = "verbose", default = FALSE),
+      makeLogicalLearnerParam(id = "covariogram", default = FALSE),
+      makeLogicalLearnerParam(id = "PR", default = FALSE),
+      makeIntegerLearnerParam(id = "pseudo", default = -1),
+      # gstat::fit.variogram params
+      makeLogicalLearnerParam(id = "fit.sills", default = TRUE),
+      makeLogicalLearnerParam(id = "fit.ranges", default = TRUE),
+      makeIntegerLearnerParam(id = "fit.method", default = 7),
+      makeLogicalLearnerParam(id = "warn.if.neg", default = FALSE),
+      makeLogicalLearnerParam(id = "fit.kappa", default = FALSE),
+      # gstat::vgm params
+      makeUntypedLearnerParam(id = "psill", default = NA),
+      makeUntypedLearnerParam(id = "model"), # FIXME what should be the type ?
+      makeUntypedLearnerParam(id = "range", default = NA),
+      makeNumericLearnerParam(id = "kappa", default = 0.5),
+      makeUntypedLearnerParam(id = "nugget"),
+      makeUntypedLearnerParam(id = "add.to"),
+      makeUntypedLearnerParam(id = "covtable"),
+      makeNumericLearnerParam(id = "Err", default = 0)
+    ),
+    par.vals = list(locations = ~x+y, nmax = 0, nmin = 0, omax = 0, maxdist = Inf,
+      dummy = FALSE, fill.all = FALSE, fill.cross = TRUE, variance = "identity", degree = 0, vdist = FALSE,
+      alpha = 0, beta.variogram = 0, cressie = FALSE, dX = 0, boundaries = 0, cloud = FALSE,
+      trend.beta = NULL, debug.level = 1, cross = TRUE, map = FALSE, projected = TRUE, lambda = 1.0,
+      verbose = FALSE, covariogram = FALSE, PR = FALSE, pseudo = -1,
+      fit.sills = TRUE, fit.ranges = TRUE, fit.method = 7, warn.if.neg = FALSE, fit.kappa = FALSE,
+      psill = NA, range = NA, kappa = 0.5, Err = 0),
```

Comment:
You don't need to list the defaults again here; only if they are different from the default (with an explanation) and parameters you didn't specify in the list above but that need to be set.

Reply:
Ok. Could you than briefly explain me the purpose of par.vals ?

Reply:
So, what is the exact purpose of par.vals ?